From 3baf6e14e056d3b59288af505fc1eafd3d4c74d1 Mon Sep 17 00:00:00 2001
Message-Id: <3baf6e14e056d3b59288af505fc1eafd3d4c74d1.1691350701.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1691350701.git.sharlatanus@gmail.com>
References: <cover.1691350701.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 6 Aug 2023 14:55:39 +0100
Subject: [PATCH 4/9] gnu: Add go-github-com-mreiferson-go-svc.

* gnu/packages/golang.scm (go-github-com-mreiferson-go-svc): New variable.
---
 gnu/packages/golang.scm | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index ce46e02158..7890e4d583 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -6256,6 +6256,34 @@ (define-public go-github-com-mr-tron-base58
 encoding and 8 times faster decoding.")
       (license license:expat))))
 
+(define-public go-github-com-mreiferson-go-svc
+  ;; NSQ specific fork of github.com/judwhite/go-svc, as Guix go build system
+  ;; does not support go.mod with `replace' statement.
+  (let ((commit "7a96e00010f68d9436e3de53a70c53f209a0c244")
+        (revision "0"))
+    (package
+      (name "go-github-com-mreiferson-go-svc")
+      (version (git-version "1.2.1" revision commit))
+      (source (origin
+                (method git-fetch)
+                (uri (git-reference
+                      (url "https://github.com/mreiferson/go-svc")
+                      (commit commit)))
+                (file-name (git-file-name name version))
+                (sha256
+                 (base32
+                  "1acgb0n3svhnraqj1fz5qc5n3b4vc5ffwyk9vfi6gcfkibm0hgmd"))))
+      (build-system go-build-system)
+      (arguments
+       '(#:import-path "github.com/judwhite/go-svc"))
+      (propagated-inputs (list go-golang-org-x-sys))
+      (home-page "https://github.com/mreiferson/go-svc")
+      (synopsis "Go Windows Service wrapper that plays nice with Linux")
+      (description
+       "Go Windows Service wrapper that plays nice with Linux.  Windows tests
+@@url{https://github.com/judwhite/go-svc/raw/master/svc/svc_windows_test.go,here}.")
+      (license license:expat))))
+
 (define-public go-github-com-gxed-hashland-keccakpg
   (let ((commit "d9f6b97f8db22dd1e090fd0bbbe98f09cc7dd0a8")
         (revision "0"))
-- 
2.40.1

